libraries/Update/src/HttpsOTAUpdate.cpp

```diff
@@ -0,0 +1,80 @@
+/* Cloud task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include"HttpsOTAUpdate.h"
+#include"Esp.h"
+
+static TaskHandle_t ota_task_handle = NULL;
+const char * server_cert;
+const char * ota_url;
+bool skip_cert_common_name_check;
+void (*http_func_ptr)(HttpEvent_t*);
+bool ota_successful = false;
+
+esp_err_t _http_event_handler(esp_http_client_event_t *evt)
+{
+    http_func_ptr(evt); 
+    return ESP_OK;
+}
+
+void aws_iot_task(void *param)
+{
+    log_i("Starting OTA example");
+    log_i("Attempting to connect to the server.........");
+
+    esp_http_client_config_t config;
+    config.url = ota_url;
+    config.cert_pem = server_cert;
+    config.skip_cert_common_name_check = skip_cert_common_name_check;
+    config.event_handler = _http_event_handler;
+    esp_err_t ret = esp_https_ota(&config);
+    if(ret == ESP_OK){
+        ota_successful = true;
+    } else {
+        log_e("Firmware upgrade failed");
+        if(ota_task_handle != NULL) {
+            vTaskDelete(ota_task_handle);
+        }
+    }
+    while (1) {
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
```

Comment:
what happened to deleting the task?

Reply:
I think that the following is enough for this, though I am not sure why you need a task for that at all. Arduinos are used to blocking API, so running the function directly is totally acceptable. It's even fine if the callback will run in the user task.

Reply:
Without creating the task the code fails to initialise HTTP connection.

Reply:
The main task stack size may not sufficient to establish HTTPs connection.

Reply:
then lets increase the main task size. Holding the memory for this one in no way makes it better. Plus you do not exit the task, but instead loop forever. If you want to use your own task, exit it when done.

Reply:
here is how I see this function:
```cpp
void https_ota_task(void *param)
 {
     esp_err_t ret = esp_https_ota((const esp_http_client_config_t *)param);
     if(ret == ESP_OK){
         status = HTTPS_OTA_SUCCESS;
     } else {
         status = HTTPS_OTA_FAIL;
     }
     vTaskDelete(NULL);
}
```

Reply:
You should use `xEventGroupSetBits` and `xEventGroupGetBits` to handle the status change and getter. One could also wait (with timeout) for a result to come.

Reply:
> You should use `xEventGroupSetBits` and `xEventGroupGetBits` to handle the status change and getter. One could also wait (with timeout) for a result to come.

Yes Required changes are made in new commit